src/main/java/net/earthcomputer/clientcommands/features/ClientTimeModifier.java

```diff
@@ -0,0 +1,34 @@
+package net.earthcomputer.clientcommands.features;
+
+public class ClientTimeModifier {
+
+    public static enum Type {
+        NONE, OFFSET, LOCK;
+    }
+
+    private static Type type = Type.NONE;
+    private static long value = 0;
+
+    public static void none() {
+        type = Type.NONE;
+    }
+
+    public static void offset(long time) {
+        type = Type.OFFSET;
+        value = time;
+    }
+
+    public static void lock(long time) {
+        type = Type.LOCK;
+        value = time;
+    }
+
+    public static long getModifiedTime(long timeOfDay) {
+        return switch (type) {
+            case NONE -> timeOfDay;
+            case LOCK -> value;
+            case OFFSET -> timeOfDay < 0 ? timeOfDay - value : timeOfDay + value;
```

Comment:
Would it make more sense to provide these implementations on the enum members themselves?

Reply:
Tbh this makes no difference in my opinion as the enum already is declared in the same class, but I moved it nontheless for you ^^